File: TableGames.Domain/UserSettings.cs
namespace TableGames.Domain
{
    public class UserSettings
    {
        public bool JoinTableAfterRoomEntered { get; set; }
        public bool JoinTableAfterTableCreated { get; set; }

        public UserSettings() {
            JoinTableAfterRoomEntered = true;
            JoinTableAfterTableCreated = true;
        }

        public object ToClient() {
            return new {
                joinTableAfterRoomEntered = JoinTableAfterRoomEntered,
                joinTableAfterTableCreated = JoinTableAfterTableCreated
            };
        }
    }
}

File: TableGames.Web/TG.Scripts/userSettings.js
define(['knockout'], function(ko) {

    function UserSettings(userSettingsState) {
        var self = this;

        self.joinTableAfterRoomEntered = ko.observable(userSettingsState.joinTableAfterRoomEntered);
        self.joinTableAfterTableCreated = ko.observable(userSettingsState.joinTableAfterTableCreated);
    }

    UserSettings.default = new UserSettings({});

    return UserSettings;
});
